ledger-service/http-json-cli/src/main/scala/com/daml/http/dbbackend/DiagnosticConfig.scala

```diff
@@ -0,0 +1,15 @@
+// Copyright (c) 2023 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.http
+package dbbackend
+
+import com.daml.dbutils
+
+import java.io.File
+
+private[http] final case class DiagnosticConfig(
+    baseConfig: dbutils.JdbcConfig,
+    clientInfo: Option[String],
```

Comment:
It is possible for Postgres as well only you would need to configure `clientInfo` and `query` differently using property and query crafted for Postgres. No need to specialised here per JDBC-driver general configuration options can be used by any jdbc driver

Reply:
I mean that AFAICT the only legal clientInfo name that Postgres understands is `"ApplicationName"`, which seems to be a session scoped thing, i.e. it is not useful for this sort of per-query diagnostics.

https://github.com/pgjdbc/pgjdbc/blob/master/pgjdbc/src/main/java/org/postgresql/jdbc/PgConnection.java#L1547

And for Oracle the only name you can use for this intended purpose (i.e. per-query diagnostics), seems to be `"OCSID.CLIENTID"`. My point is that we're using it for a specific purpose, for which there's only one useful value in Oracle, and no useful values in Postgres, so allowing any arbitrary string here seems like a poor interface.

Reply:
Can initialise `diagnosticsClientInfo` to None by default. Further distinction per concrete JDBC driver is not needed IMHO. What we are really talking about here is `Coding to interfaces, not implementation.` The code atm is using interface https://docs.oracle.com/javase/8/docs/api/java/sql/Connection.html#setClientInfo-java.util.Properties- rather than concrete PG implementation details. I didn't even check what is possible atm on PG as that is not pressing issue for us now. The fact that PG currently implements the interface in rather limited way doesn't mean next version of PG JDBC cannot change that. But I doubt that `java.sql.Connection` as interface will change any time soon.

Reply:
My point is that I think it's coding to the _wrong_ interface. The `setClientInfo` is very broad and loosely-defined, by the fact that it accepts an arbitrary string as the name. We have a very specific intent with our diagnostics here, which is to populate information that allows us to trace and diagnose individual queries. As such, you can't just pass in any string there and have it do the right thing. The interface for our case is actually very constrained, and differs per choice of db, so by exposing an arbitrary string here we're only unnecessarily introducing all kinds of ways for the user to get this wrong.

I like this as a general guideline for assessing APIs/interfaces
http://sweng.the-davies.net/Home/rustys-api-design-manifesto

With regard to the `clientInfo` setting specifically, having the field present in the config puts it at a 2 or 3. Having it absent, and populated automatically from the driver puts it at a 10.